### ISSUE TITLE
TypeError: invalid options argument after 2.18

### ISSUE THREAD

**Issue by daniluk4000** [nuxt/nuxt#27873]
### Environment

  System:
    OS: Linux 5.15 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12700K
    Memory: 14.03 GB / 15.50 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 22.1.0 - /usr/bin/node
    Yarn: 4.3.1 - /usr/bin/yarn
    npm: 10.7.0 - /usr/bin/npm
    pnpm: 9.1.1 - /usr/bin/pnpm
  npmPackages:
    @nuxt/types: ^2.18.0 => 2.18.0
    @nuxt/typescript-build: ^3.0.2 => 3.0.2
    nuxt: ^2.18.0 => 2.18.0

### Reproduction

https://github.com/daniluk4000/rip-nuxt2

### Describe the bug

After upgrade to Nuxt 2.16 it started showing huge errors in our environment (also in CI)

### Logs

```shell-script
/WebstormProjects/nuxt2/node_modules/@nuxt/webpack/node_modules/mkdirp/lib/opts-arg.js:13
    throw new TypeError('invalid options argument')
    ^

TypeError: invalid options argument
    at optsArg (/WebstormProjects/nuxt2/node_modules/@nuxt/webpack/node_modules/mkdirp/lib/opts-arg.js:13:11)
    at Function.mkdirp (/WebstormProjects/nuxt2/node_modules/@nuxt/webpack/node_modules/mkdirp/index.js:11:10)
    at /WebstormProjects/nuxt2/node_modules/webpack/lib/Compiler.js:494:26
    at eval (eval at create (/WebstormProjects/nuxt2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at /WebstormProjects/nuxt2/node_modules/@nuxt/webpack/dist/webpack.js:1664:7
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/WebstormProjects/nuxt2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:12:1)
    at AsyncSeriesHook.lazyCompileHook (/WebstormProjects/nuxt2/node_modules/tapable/lib/Hook.js:154:20)
    at Compiler.emitAssets (/WebstormProjects/nuxt2/node_modules/webpack/lib/Compiler.js:491:19)
    at onCompiled (/WebstormProjects/nuxt2/node_modules/webpack/lib/Watching.js:51:19)
    at /WebstormProjects/nuxt2/node_modules/webpack/lib/Compiler.js:681:15
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/WebstormProjects/nuxt2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at AsyncSeriesHook.lazyCompileHook (/WebstormProjects/nuxt2/node_modules/tapable/lib/Hook.js:154:20)
    at /WebstormProjects/nuxt2/node_modules/webpack/lib/Compiler.js:678:31
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/WebstormProjects/nuxt2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at AsyncSeriesHook.lazyCompileHook (/WebstormProjects/nuxt2/node_modules/tapable/lib/Hook.js:154:20)
    at /WebstormProjects/nuxt2/node_modules/webpack/lib/Compilation.js:1423:35
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/WebstormProjects/nuxt2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at AsyncSeriesHook.lazyCompileHook (/WebstormProjects/nuxt2/node_modules/tapable/lib/Hook.js:154:20)
    at /WebstormProjects/nuxt2/node_modules/webpack/lib/Compilation.js:1414:32
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/WebstormProjects/nuxt2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at AsyncSeriesHook.lazyCompileHook (/WebstormProjects/nuxt2/node_modules/tapable/lib/Hook.js:154:20)
    at /WebstormProjects/nuxt2/node_modules/webpack/lib/Compilation.js:1409:36
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/WebstormProjects/nuxt2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at AsyncSeriesHook.lazyCompileHook (/WebstormProjects/nuxt2/node_modules/tapable/lib/Hook.js:154:20)
    at /WebstormProjects/nuxt2/node_modules/webpack/lib/Compilation.js:1405:32
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/WebstormProjects/nuxt2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at AsyncSeriesHook.lazyCompileHook (/WebstormProjects/nuxt2/node_modules/tapable/lib/Hook.js:154:20)
    at Compilation.seal (/WebstormProjects/nuxt2/node_modules/webpack/lib/Compilation.js:1342:27)
    at /WebstormProjects/nuxt2/node_modules/webpack/lib/Compiler.js:675:18
    at /WebstormProjects/nuxt2/node_modules/webpack/lib/Compilation.js:1261:4
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/WebstormProjects/nuxt2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:20:1)
    at AsyncSeriesHook.lazyCompileHook (/WebstormProjects/nuxt2/node_modules/tapable/lib/Hook.js:154:20)
    at Compilation.finish (/WebstormProjects/nuxt2/node_modules/webpack/lib/Compilation.js:1253:28)
    at /WebstormProjects/nuxt2/node_modules/webpack/lib/Compiler.js:672:17
    at eval (eval at create (/WebstormProjects/nuxt2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:11:1)
    at /WebstormProjects/nuxt2/node_modules/webpack/lib/Compilation.js:1185:12
    at /WebstormProjects/nuxt2/node_modules/webpack/lib/Compilation.js:1097:9
    at process.processTicksAndRejections (node:internal/process/task_queues:77:11)
```

**Comment by danielroe** [comment]
Could you see what version of mkdirp is in your lockfile? 🙏

**Comment by daniluk4000** [comment]
> mkdirp

![image](https://github.com/nuxt/nuxt/assets/10450717/bb682836-cdff-44ab-981e-4ca74facb4a5)

It's showing those two. The one in node_modules is 0.x...

But

```
├─ @nuxt/webpack@npm:2.18.0
│  └─ mkdirp@npm:1.0.4 (via npm:^1.0.4)
```

resolutions for 1.0.4 didn't help

**Comment by everyx** [comment]
A workaround for npm users:

```typescript
  "overrides": {
    "nuxt": {
      "@nuxt/webpack": {
        "mkdirp": "0.5.6"
      }
    }
  }
```

**Comment by danielroe** [comment]
this should now be resolved in [v2.18.1](https://github.com/nuxt/nuxt/releases/tag/v2.18.1).
